### PR TITLE
Feature/examples debug

### PR DESCRIPTION
**Pull request by andre-merzky**
This adds very simple usage examples for some debug utilities.  It also switches from float based random specs for `raise_on` to int-based ones, which are easier IMHO.

### CODE REVIEW

File: src/radical/utils/debug.py [hunk 1]
```diff
@@ -281,8 +281,13 @@ def get_caller_name(skip=2):
 
 # ------------------------------------------------------------------------------
 #
+_verb  = False
+if 'RADICAL_DEBUG' in os.environ :
+    _verb = True
```
Comment (iparask) [inline src/radical/utils/debug.py]: There is an assumption here that whenever `RADICAL_DEBUG` exists in the environment it is going to be `True`. What if it is `False` or something else?

File: src/radical/utils/tracer.py [hunk 2]
```diff
@@ -49,11 +49,13 @@ def _tracer (frame, event, arg) :
 
         filename = frame.f_globals["__file__"]
         lineno   = frame.f_lineno
+
```
Comment (iparask) [inline src/radical/utils/tracer.py]: I remember you did not like more than one empty lines?! :)

File: src/radical/utils/debug.py (hunk 1 above)
Comment (andre-merzky) [reply src/radical/utils/debug.py]: RADICAL_DEBUG can be empty or set to anything - it just needs to be present.  I wanted this to be a quick check (no additional string compares etc).

File: src/radical/utils/tracer.py (hunk 2 above)
Comment (andre-merzky) [reply src/radical/utils/tracer.py]: Ha :)

**Pull request merged**